model-datastructure/src/commonMain/kotlin/org/modelix/model/LinearHistory.kt

```diff
@@ -5,6 +5,10 @@ import org.modelix.model.lazy.IDeserializingKeyValueStore
 import org.modelix.model.lazy.KVEntryReference
 import org.modelix.model.persistent.CPVersion
 
+/**
+ * When merging two branches there might be conflicts between operations that need to be resolved in a deterministic
+ * way. This class puts them in an order that preserves the order of earlier merges.
```

Comment:
Can you give some ideas or references for the algorithm that achieves this property?